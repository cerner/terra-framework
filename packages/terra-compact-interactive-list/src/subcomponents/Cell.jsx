import React, { useContext, useState, useEffect } from 'react';
import * as KeyCode from 'keycode-js';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import { checkIfColumnIsResponsive } from '../utils/utils';
import getFocusableElements from '../../../terra-table/src/utils/focusManagement';
import formattedColumnShape from '../proptypes/formattedColumnShape';
import { widthUnitTypes, alignTypes } from '../utils/constants';
import styles from './Cell.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * String identifier of the row in which the Cell will be rendered.
   */
  rowId: PropTypes.string.isRequired,

  /**
   * Boolean indicating that the cell is a row header.
   */
  isRowHeader: PropTypes.bool,

  /**
   * Content that will be rendered within the Cell.
   */
  children: PropTypes.node,

  /**
   * A column data.
   */
  column: formattedColumnShape,

  /**
   * Columns minimum width in units set by widthUnit prop.
   */
  columnMinimumWidth: PropTypes.number,

  /**
   * Columns maximum width in units set by widthUnit prop.
   */
  columnMaximumWidth: PropTypes.number,

  /**
   * The type of width value. Defaults to px.
   */
  widthUnit: PropTypes.oneOf([
    widthUnitTypes.PX,
    widthUnitTypes.EM,
    widthUnitTypes.REM,
  ]).isRequired,

  /**
   * Callback function that will be called on click on the cell.
   */
  onCellSelect: PropTypes.func,

  /**
   * Boolean indicating whether the Cell is currently selected.
   */
  isSelected: PropTypes.bool,

  /**
   * Callback function that will pass the focused column and focused row ids to the main component.
   */
  setFocusedCell: PropTypes.func,
};

const Cell = (props) => {
  const {
    rowId,
    children,
    column,
    columnMinimumWidth,
    columnMaximumWidth,
    widthUnit,
    onCellSelect,
    isSelected,
    setFocusedCell,
    isRowHeader,
  } = props;

  const theme = useContext(ThemeContext);
  const cellRef = React.useRef();
  const [isSelectableCell, setIsSelectableCell] = useState(false);
  const selectable = column.isSelectable && isSelectableCell;
  const className = cx('cell', theme.className, {
    selected: selectable && isSelected,
    selectable,
  });

  const {
    id,
    width,
    flexGrow,
    maximumWidth,
    minimumWidth,
    align,
  } = column;

  const isResponsive = checkIfColumnIsResponsive(flexGrow, width);

  useEffect(() => {
    if (cellRef?.current) {
      const interactiveChildren = getFocusableElements(cellRef?.current);
      // if no interactive elements
      if (interactiveChildren?.length === 0) {
        setIsSelectableCell(true);
      }
    }
  }, [cellRef]);

  const handleKeyDown = (event) => {
    // does not need setFocusedCell call as the cell focused by key is tracked in CompactInteractiveList component
    const key = event.keyCode;
    if (key === KeyCode.KEY_SPACE && isSelectableCell) {
      event.preventDefault(); // prevents scroll on empty cells
      if (onCellSelect) {
        onCellSelect({ rowId, columnId: id });
      }
    }
  };

  const handleMouseDown = (event) => {
    setFocusedCell({ rowId, columnId: id });
    if (isSelectableCell) {
      if (onCellSelect) {
        onCellSelect({ rowId, columnId: id });
      }
    } else {
      // allows clickable elements inside non-nselactable cells to be clicked, but not the cell itself
      event.preventDefault();
    }
  };

  const style = {
    justifyContent: align || alignTypes.LEFT,
    textAlign: align || alignTypes.LEFT,
  };
  if (isResponsive) {
    // flex and minimumWidth props required for responsive cells
    style.flex = `1 1 ${width || minimumWidth || columnMinimumWidth}${widthUnit}`;
    style.minWidth = `${(minimumWidth || columnMinimumWidth)}${widthUnit}`;
    // assign maxWidth prop only if such exist
    if ((maximumWidth || columnMaximumWidth)) {
      style.maxWidth = `${(maximumWidth || columnMaximumWidth)}${widthUnit}`;
    }
  } else {
    // width prop is required for fixed width cells
    style.width = `${width}${widthUnit}`;
  }

  return (
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    <div
      role={isRowHeader ? 'rowheader' : 'gridcell'}
      ref={cellRef}
      className={className}
      tabIndex={isSelectableCell ? -1 : null}
      // eslint-disable-next-line react/forbid-dom-props
      style={style}
      onMouseDown={handleMouseDown}
      onKeyDown={handleKeyDown}
    >
      {children}
    </div>
  );
};

Cell.propTypes = propTypes;

export default React.memo(Cell);
