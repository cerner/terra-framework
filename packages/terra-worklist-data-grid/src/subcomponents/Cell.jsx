/* eslint-disable react/forbid-dom-props */
import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import * as KeyCode from 'keycode-js';
import classNames from 'classnames/bind';

import ThemeContext from 'terra-theme-context';

import WorklistDataGridUtils from '../utils/WorklistDataGridUtils';
import styles from './Cell.module.scss';
import '../_elementPolyfill';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * String identifier of the row in which the Cell will be rendered.
   */
  rowId: PropTypes.string.isRequired,

  /**
   * String identifier of the column in which the Cell will be rendered.
   */
  columnId: PropTypes.string.isRequired,

  /**
   * The cell's row position in the grid. This is zero based.
   */
  rowIndex: PropTypes.number,

  /**
   * The cell's column position in the grid. This is zero based.
   */
  columnIndex: PropTypes.number,

  /**
   * Content that will be rendered within the Cell.
   */
  children: PropTypes.node,

  /**
   *  Boolean indicating if cell contents are masked.
   */
  isMasked: PropTypes.bool,

  /**
   *  Boolean indicating if cell is pinned.
   */
  isPinned: PropTypes.bool,

  /**
   * Boolean value indicating whether or not the column header is selectable.
   */
  isSelectable: PropTypes.bool,

  /**
   * Boolean value to indicate if the cell is the tab stop on the grid. At any given time, the grid has only one tab stop.
   */
  isTabStop: PropTypes.bool,

  /**
   * Boolean indicating whether the Cell is currently selected.
   */
  isSelected: PropTypes.bool,

  /**
   * String that labels the cell for accessibility.
   */
  ariaLabel: PropTypes.string,

  /**
   * Boolean indicating that the cell is a row header.
   */
  isRowHeader: PropTypes.bool,

  /**
   * Callback function that will be called when this cell is selected.
   */
  onCellSelect: PropTypes.func,

  /**
   * String that specifies the height of the cell. Any valid CSS value is accepted.
   */
  height: PropTypes.string,

  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
};

const defaultProps = {
  isTabStop: false,
  isRowHeader: false,
  isSelected: false,
  isSelectable: true,
  isMasked: false,
  isPinned: false,
};

function Cell(props) {
  const {
    rowId,
    columnId,
    rowIndex,
    columnIndex,
    isTabStop,
    ariaLabel,
    isMasked,
    isPinned,
    isRowHeader,
    isSelectable,
    isSelected,
    children,
    onCellSelect,
    height,
    intl,
  } = props;

  const theme = useContext(ThemeContext);

  const handleMouseDown = (event) => {
    if (isMasked || !isSelectable) {
      event.stopPropagation();
      event.preventDefault();
    } else if (onCellSelect) {
      onCellSelect({ rowId, columnId }, { row: rowIndex, col: columnIndex });
      event.stopPropagation();
    }
  };

  const handleKeyDown = (event) => {
    const key = event.keyCode;
    switch (key) {
      case KeyCode.KEY_SPACE:
        if (isMasked || !isSelectable) {
          event.stopPropagation();
          event.preventDefault();
        } else if (onCellSelect) {
          onCellSelect({ rowId, columnId }, { row: rowIndex, col: columnIndex });
          event.stopPropagation();
          event.preventDefault(); // prevent the default scrolling
        }
        break;
      case KeyCode.KEY_C:
        if (event.ctrlKey || event.metaKey) {
          WorklistDataGridUtils.writeToClipboard(event.target.textContent);
        }
        break;
      default:
    }
  };

  let cellAriaLabel = ariaLabel;
  if (isMasked) {
    cellAriaLabel = intl.formatMessage({ id: 'Terra.worklistDataGrid.maskedCell' });
  } else if (!children) {
    cellAriaLabel = intl.formatMessage({ id: 'Terra.worklistDataGrid.blank' });
  }

  const className = cx('worklist-data-grid-cell', {
    masked: isMasked,
    pinned: isPinned,
    selectable: isSelectable && !isMasked,
    selected: isSelected && !isMasked,
    blank: !children,
  }, theme.className);

  const CellTag = isRowHeader ? 'th' : 'td';
  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
    <CellTag
      aria-selected={isSelected}
      aria-label={cellAriaLabel}
      tabIndex={isTabStop ? 0 : -1}
      className={className}
      onMouseDown={handleMouseDown}
      onKeyDown={handleKeyDown}
    >
      {!isMasked && children && <div className={cx('cell-content', theme.className)} style={{ height }}>{children}</div>}
    </CellTag>
  );
}

Cell.propTypes = propTypes;
Cell.defaultProps = defaultProps;

export default injectIntl(Cell);
