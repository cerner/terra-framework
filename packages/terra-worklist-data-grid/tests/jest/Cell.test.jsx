import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import Cell from '../../src/subcomponents/Cell';

describe('Cell', () => {
  it('renders a row header cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        isTabStop={false}
        isSelected={false}
        cell={{ content: 'Data in cell' }}
        isRowHeader
        onCellSelect={jest.fn}
      />,
    ).dive();

    expect(wrapper.find('th')).toHaveLength(1); // Th when row header.
    expect(wrapper.find('td')).toHaveLength(0); // There should be no th since this is row header.
    expect(wrapper).toMatchSnapshot();

    const cellContent = wrapper.find('th').find('div');
    expect(cellContent).toHaveLength(1);
    expect(cellContent.text()).toEqual('Data in cell');
  });

  it('renders a row header cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        isTabStop={false}
        isSelected={false}
        cell={{ content: 'Data in cell' }}
        isRowHeader
        onCellSelect={jest.fn}
      />,
    ).dive();

    expect(wrapper.find('th')).toHaveLength(1); // Th when row header.
    expect(wrapper.find('td')).toHaveLength(0); // There should be no th since this is row header.
    expect(wrapper).toMatchSnapshot();

    const cellContent = wrapper.find('th').find('div');
    expect(cellContent).toHaveLength(1);
    expect(cellContent.text()).toEqual('Data in cell');
  });

  it('renders a regular(non row header) cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        isTabStop={false}
        isSelected={false}
        cell={{ content: 'Data in cell' }}
        isRowHeader={false}
        onCellSelect={jest.fn}
      />,
    ).dive();

    const normalCell = wrapper.find('td');
    expect(normalCell).toHaveLength(1); // Th when non row header.
    expect(wrapper.find('th')).toHaveLength(0); // There should be no td since this is a row header.

    const cellContent = normalCell.find('div');
    expect(cellContent).toHaveLength(1);
    expect(cellContent.text()).toEqual('Data in cell');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a selectable cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        cell={{ isSelectable: true, content: 'Data in cell' }}
        onCellSelect={jest.fn}
      />,
    ).dive();

    const selectableCell = wrapper.find('td.selectable');
    expect(selectableCell).toHaveLength(1); // Cell should have been styled selectable.

    const cellContent = selectableCell.find('div');
    expect(cellContent).toHaveLength(1);
    expect(cellContent.text()).toEqual('Data in cell');

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a blank cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        cell={{ content: '' }}
        onCellSelect={jest.fn}
      />,
    ).dive();

    const blankCell = wrapper.find('td.blank');
    expect(blankCell).toHaveLength(1);

    const cellContent = blankCell.find('div');
    expect(cellContent).toHaveLength(0);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a selected cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        ariaLabel="Some Label Here"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        isSelected
        cell={{ content: 'Data in cell' }}
        onCellSelect={jest.fn}
      />,
    ).dive();

    const selectedCell = wrapper.find('td.selected');
    expect(selectedCell).toHaveLength(1); // Cell should have been styled selected.

    expect(wrapper.find('td').prop('aria-selected')).toBe(true);
    expect(wrapper.find('td').prop('aria-label')).toBe('Some Label Here');

    const cellContent = selectedCell.find('div');
    expect(cellContent).toHaveLength(1);
    expect(cellContent.text()).toEqual('Data in cell');

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a mask cell', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        ariaLabel="Some Label Here"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        cell={{ isMasked: true, content: 'Data in cell' }}
        onCellSelect={jest.fn}
      />,
    ).dive();

    const maskedCell = wrapper.find('td.masked');
    expect(maskedCell).toHaveLength(1);

    expect(wrapper.find('td').prop('aria-label')).toBe('Terra.worklistDataGrid.maskedCell');

    const cellContent = maskedCell.find('div');
    expect(cellContent).toHaveLength(0); // Masked cell should have no content.

    expect(wrapper).toMatchSnapshot();
  });

  it('renders cell as mask only even if cell is selectable and selected', () => {
    const wrapper = shallowWithIntl(
      <Cell
        rowId="RowID"
        columnId="ColumnId"
        ariaLabel="Some Label Here"
        coordinates={{ row: 1, col: 2 }}
        key="key"
        isSelected
        cell={{
          isMasked: true, isSelectable: true, content: 'Data in cell',
        }}
        onCellSelect={jest.fn}
      />,
    ).dive();

    const maskedCell = wrapper.find('td.masked');
    expect(maskedCell).toHaveLength(1);

    expect(maskedCell.prop('aria-label')).toBe('Terra.worklistDataGrid.maskedCell');

    const cellContent = maskedCell.find('div');
    expect(cellContent).toHaveLength(0); // Masked cell should have no content.

    // Verify other styles not applied.
    expect(wrapper.find('td.selected')).toHaveLength(0); // Cell should not be styled selected since it is masked.
    expect(wrapper.find('td.selectable')).toHaveLength(0); // Cell should not be styled selectable since it is masked.

    expect(wrapper).toMatchSnapshot();
  });
});

