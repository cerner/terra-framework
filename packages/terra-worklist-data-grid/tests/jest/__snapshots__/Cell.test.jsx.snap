// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders a pinned cell 1`] = `
<InjectIntl(Cell)
  columnId="ColumnId"
  columnIndex={0}
  height="50px"
  isRowHeader={true}
  isSelected={false}
  isTabStop={false}
  key="key"
  onCellSelect={[Function]}
  rowId="RowID"
  rowIndex={1}
>
  <Cell
    columnId="ColumnId"
    columnIndex={0}
    height="50px"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    isMasked={false}
    isRowHeader={true}
    isSelectable={true}
    isSelected={false}
    isTabStop={false}
    onCellSelect={[Function]}
    rowId="RowID"
    rowIndex={1}
  >
    <th
      aria-selected={false}
      className="worklist-data-grid-cell pinned selectable"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="rowheader"
      scope="row"
      style={
        Object {
          "left": 0,
        }
      }
      tabIndex={-1}
    >
      <FocusTrap
        _createFocusTrap={[Function]}
        active={false}
        focusTrapOptions={
          Object {
            "clickOutsideDeactivates": true,
            "escapeDeactivates": false,
            "onDeactivate": [Function],
            "returnFocusOnDeactivate": true,
          }
        }
        paused={false}
      >
        <div
          className="cell-content"
          style={
            Object {
              "height": "50px",
            }
          }
        >
          Pinned cell data
        </div>
      </FocusTrap>
    </th>
  </Cell>
</InjectIntl(Cell)>
`;

exports[`Cell verifies mask takes precedence when cell is masked, selectable and selected 1`] = `
<td
  aria-label="Some Label Here"
  aria-selected={true}
  className="worklist-data-grid-cell masked"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <span
        className="no-data-cell"
      >
        Terra.worklistDataGrid.maskedCell
      </span>
    </div>
  </FocusTrap>
</td>
`;

exports[`Cell verifies that a cell has the correct styles and no content when isMasked prop is true 1`] = `
<td
  aria-label="Some Label Here"
  aria-selected={false}
  className="worklist-data-grid-cell masked"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <span
        className="no-data-cell"
      >
        Terra.worklistDataGrid.maskedCell
      </span>
    </div>
  </FocusTrap>
</td>
`;

exports[`Cell verifies that a selected cell has the correct styles when isSelected prop is true 1`] = `
<td
  aria-label="Some Label Here"
  aria-selected={true}
  className="worklist-data-grid-cell selectable selected"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      Data in cell
    </div>
  </FocusTrap>
</td>
`;

exports[`Cell verifies that only a regular cell(<td>) is created when isRowHeader prop is false 1`] = `
<td
  aria-selected={false}
  className="worklist-data-grid-cell selectable"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      Data in cell
    </div>
  </FocusTrap>
</td>
`;

exports[`Cell verifies that only a row header cell (<th>) is created when isRowHeader prop is true 1`] = `
<th
  aria-selected={false}
  className="worklist-data-grid-cell selectable"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="rowheader"
  scope="row"
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": "50px",
        }
      }
    >
      Data in cell
    </div>
  </FocusTrap>
</th>
`;

exports[`Cell verifies that the cell rendered is marked blank when the cell content is empty 1`] = `
<td
  aria-selected={false}
  className="worklist-data-grid-cell selectable blank"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <span
        className="no-data-cell"
      >
        Terra.worklistDataGrid.blank
      </span>
    </div>
  </FocusTrap>
</td>
`;

exports[`Cell verifies that the cell rendered is marked selectable when isSelectable prop is true 1`] = `
<td
  aria-selected={false}
  className="worklist-data-grid-cell selectable"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "left": null,
    }
  }
  tabIndex={-1}
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={false}
    focusTrapOptions={
      Object {
        "clickOutsideDeactivates": true,
        "escapeDeactivates": false,
        "onDeactivate": [Function],
        "returnFocusOnDeactivate": true,
      }
    }
    paused={false}
  >
    <div
      className="cell-content"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      Data in cell
    </div>
  </FocusTrap>
</td>
`;
